MetadataMigration/src/main/java/org/opensearch/migrations/commands/MigratorEvaluatorBase.java

```diff
@@ -137,4 +138,10 @@ private IndexMetadataResults migrateIndices(MigrationMode mode, Clusters cluster
         var indexResults = indexRunner.migrateIndices(mode, context.createIndexContext());
         log.info("Index copy complete.");
         return indexResults;
-    } }
+    }
+
+    protected String createUnexpectedErrorMessage(Throwable e) {
+        var causeMessage = Optional.of(e).map(Throwable::getCause).map(Throwable::getMessage).orElse(null);
+        return "Unexpected failure: " + e.getMessage() + (causeMessage == null ? "" : ", inner cause: " + causeMessage);
```

Comment:
If we have *any* kind of exception, we'll NOT show a stack trace?  If there's a NullPointerException, I certainly want to see the stack trace for that.  You should scope down 'expected' vs 'unexpected' exceptions and treat them differently.

Reply:
This is presentation layer, user are not suppose to see callstacks - this is the intention.  Actual exceptions are logged and available in the metadata run log.